docs/components/InputFile/InputFile.stories.mdx

```diff
@@ -58,3 +58,33 @@ updating an array of `FileUpload`s based on their `key` as uploads progress.
  */
 export function updateFiles(updatedFile: FileUpload, files: FileUpload[]);
 ```
+
+### validator
+
+InputFile can accept a `validator` function as a prop. This will allow you to
+reject or accept file uploads based on your needs. The value of the function
+must accept a File and return `null` when it is accepted or an error object when
+rejected.
+
+For example, the code below only accepts files that are less than 2MB in size.
+
+```ts
+function pngFileValidator(file: File) {
+  if (file.size > 2000000) {
+    return {
+      code: "file-too-big",
+      message: "The file is too big. Try using a smaller image",
+    };
+  }
+
+  return null;
+}
+```
+
+#### Notes
+
+If you would like to restrict specific file types, it is advised that you use
+the `allowedTypes` property instead of the `validator` property. Using
+`allowTypes` will allow the OS know what type of files the uploader is expecting
```

Comment:
`allowedTypes` rather than `allowTypes`

I think there's a word missing here

>will allow the OS know what

but you know what I think we can just leave it at 

>If you would like to restrict specific file types, it is advised that you use
the `allowedTypes` property instead of the `validator` property.

that gets at the main thing we want to communicate. describing all the behavior of the `allowedTypes` would be better suited either in its own section or on the prop JS docs